### DISCUSSION TITLE
How to paste git token?

### DISCUSSION THREAD

**Discussion by TechnoLukas** [jesseduffield/lazygit#3225] [Q&A]
Hello. 

When I try to fetch or pull I get this popup with Nickname and password. 
When i enter everything i get this...
![image](https://github.com/jesseduffield/lazygit/assets/110934679/bb1e6e13-86fe-4325-ae10-9cae77f6084b)
For some reason when I click on the links i can't open them in the browser. Maybe because i use fish shell... not sure.

Following this https://dev.to/shafia/support-for-password-authentication-was-removed-please-use-a-personal-access-token-instead-4nbk guide, i created the token. 
How to put his token into the lazygit?

**Comment by stefanhaller** [comment]
Any reason why you are using https to access your repos, and not ssh?

I'd recommend to switch to ssh unless there's some reason why you can't, makes everything easier. Start reading [here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh).

**Reply by TechnoLukas** [reply]
Hmm.. ok.
1. So there is no way to work with without ssh connection?

2. I followed that guide, and created ssh key for my reposity.. How to connect to it with lazygit? When i open lazygit it still shows https.

**Reply by stefanhaller** [reply]
1. Shrug. I don't know, but I guess not. I suspect nobody cares, because nobody uses https. 😄 
2. You need to change your remote URL. In lazygit, go to the Remotes tab, select origin, press 'e' (for edit), and change the URL to start with `git@github.com`.

**Comment by pdcxs** [comment]
I cannot find a solution. But you can use git itself to specify the remote url:

Use this to see current url:

``` bash
git remote -v
```

And set the remote url:

```bash
git remote set-url origin https://new-url.com/user/repo.git
```

To verify the result:

```bash
git remote -v
```
